datamade/la-metro-councilmatic
# Merge master into upgrade and refactor autocomplete backend

## Overview

This PR brings changes from master into the upgrade. It also refactors the backend approach for autocomplete to work with the upgrade model changes (namely, there is no longer a separate Councilmatic Subject). The refactor is contained to [this commit range](https://github.com/datamade/la-metro-councilmatic/pull/532/files/59c3d1f08d1f5c4f81769cc6a08397f65a55197b..b8b9fe6b93c99a3bd03f338c0d1422d8e803b7c2). I would recommend reviewing that diff alone, since the rest has already been reviewed.

### Checklist


### Demo

N/A

### Notes

N/A

## Testing Instructions

 * `blackbox_cat configs/settings_deployment.staging.py`, then copy the `SMART_LOGIC_*` settings into your local deployment settings.
* If you don't have a local database, run a scrape: `docker-compose run --rm scrapers`.
* Run the management command to sync subjects with their GUIDs: `docker-compose run --rm app python manage.py refresh_guid`.
* Run the application and navigate to the homepage. Type `Metro Red Line` into the search box and confirm you are shown suggestions. Click `Metro Red Line` in the drop down and confirm you're redirected to the search results page and the results meet your expectations.
